gobblin-cluster/src/main/java/org/apache/gobblin/cluster/HelixAssignedParticipantCheck.java

```diff
@@ -139,16 +139,24 @@ public void execute() throws CommitStepException {
 
       if (jobContext != null) {
         String participant = jobContext.getAssignedParticipant(partitionNum);
-        if (participant != null) {
-          boolean isAssignedParticipant = participant.equalsIgnoreCase(helixInstanceName);
-          if (!isAssignedParticipant) {
-            log.info("The current helix instance is not the assigned participant. helixInstanceName={}, assignedParticipant={}",
-                helixInstanceName, participant);
-          }
-
-          return isAssignedParticipant;
+        if (participant == null) {
+          log.error("The current assigned participant is null. This implies that \n"
```

Comment:
If this is not expected, we should throw exception

Reply:
The helix assigned participant check will cause the task to fail when you return false. It will then go through all of the procedures we do when there is a container / pre commit check failure. 

In this case, it won't silently fail anyways. Do you still think it's better to throw an exception immediately?

Reply:
Oh yeah, if that's the case, I'm fine with we just return false here.